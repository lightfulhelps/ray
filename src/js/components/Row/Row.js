// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ChildrenArray<React.Element<any>>,
  className?: string,
  noGutters?: boolean,
  tag?: string,
};

const Row = ({ children, className, noGutters, tag: Tag = 'div' }: Props) => {
  const classes = classNames(className, 'row', { 'no-gutters': noGutters });

  return <Tag className={classes}>{children}</Tag>;
};

export default Row;
