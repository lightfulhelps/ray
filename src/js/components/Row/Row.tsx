import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ReactNode;
  className?: string;
  noGutters?: boolean;
  tag?: keyof JSX.IntrinsicElements;
};

const Row: React.FC<Props> = ({
  children,
  className,
  noGutters,
  tag: Tag = 'div',
  ...other
}: Props) => {
  const classes = classNames(className, 'row', { 'no-gutters': noGutters });

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default Row;
