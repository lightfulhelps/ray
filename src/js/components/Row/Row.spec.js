import React from 'react';
import { shallow } from 'enzyme';
import Row from './Row';

const setup = (overrides = {}) => {
  const props = {
    children: <p>Children</p>,
    ...overrides,
  };
  const wrapper = shallow(<Row {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Row />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the children prop', () => {
    const { wrapper } = setup();

    expect(wrapper.children().text()).toEqual('Children');
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('row')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the noGutter prop', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('no-gutters')).toBe(false);

    wrapper.setProps({ noGutters: true });

    expect(wrapper.hasClass('no-gutters')).toBe(true);
  });

  it('should handle the tag prop', () => {
    const { wrapper } = setup();

    expect(wrapper.type()).toBe('div');

    wrapper.setProps({ tag: 'a' });

    expect(wrapper.type()).toBe('a');
  });
});
