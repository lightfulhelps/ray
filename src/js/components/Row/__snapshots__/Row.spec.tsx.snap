// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should render 1`] = `
<div
  className="row"
>
  <p>
    Children
  </p>
</div>
`;
